statsmodels/regression/tests/test_recursive_ls.py

```diff
@@ -140,6 +141,94 @@ def test_ols():
+    # (the latter are defined as e_t|T by Harvey, 1989, 5.4.5)
+    # (this follows since the smoothed state simply contains the
+    # full-information estimates of the regression coefficients)
+    actual = (mod.endog[:, 0] -
+              np.sum(mod['design', 0, :, :] * res.smoothed_state, axis=0))
+    assert_allclose(actual, res_glm.resid_response)
+
+    # Given the estimate of scale as `sum(v_t^2 / f_t) / (T - d)` (see
+    # Harvey, 1989, 4.2.5 on p. 183), then llf_recursive is equivalent to the
+    # full OLS loglikelihood (i.e. without the scale concentrated out).
+    desired = mod_glm.loglike(res_glm.params, scale=res_glm.scale)
+    assert_allclose(res.llf_recursive, desired)
+    # Alternatively, we can constrcut the concentrated OLS loglikelihood
+    # by computing the scale term with `nobs` in the denominator rather than
+    # `nobs - d`.
+    scale_alternative = np.sum((
+        res.standardized_forecasts_error[0, 1:] *
+        res.filter_results.obs_cov[0, 0]**0.5)**2) / mod.nobs
+    llf_alternative = np.log(norm.pdf(res.resid_recursive, loc=0,
+                                      scale=scale_alternative**0.5)).sum()
+    assert_allclose(llf_alternative, res_glm.llf)
+
+    # Prediction
+    # TODO: prediction in this case is not working.
+    design = np.ones((2, 3, 10))
+    design[1] = mod['design', 1, :, 0:1]
+    assert_raises(NotImplementedError, res.forecast, 10, design=design)
+    # actual = res.forecast(10, design=design)
+    # assert_allclose(actual, res_glm.predict(np.ones((10, 3))))
+
+    # Hypothesis tests
+    actual = res.t_test('m1 = 0')
+    desired = res_glm.t_test('m1 = 0')
+    assert_allclose(actual.statistic, desired.statistic)
+    assert_allclose(actual.pvalue, desired.pvalue, atol=1e-15)
+
+    actual = res.f_test('m1 = 0')
+    desired = res_glm.f_test('m1 = 0')
+    assert_allclose(actual.statistic, desired.statistic)
+    # TODO: Why is this tolerance relatively wide, whereas the t_test is very
+    # tight?
+    assert_allclose(actual.pvalue, desired.pvalue, atol=1e-6)
+
+    # Information criteria
+    # Note: the llf and llf_obs given in the results are based on the Kalman
+    # filter and so the ic given in results will not be identical to the
+    # OLS versions. Additionally, llf_recursive is comparable to the
+    # non-concentrated llf, and not the concentrated llf that is by default
+    # used in OLS. Compute new ic based on llf_alternative to compare.
+    # TODO: Why do I need a -1 here but not in the unconstrained case? It must
+    # have something to do with the fact that GLM does not report the intercept
+    # as a degree of freedom so we need to subtract it here, but then why don't
+    # we need to do that it the unconstrained case?
+    actual_aic = aic(llf_alternative, res.nobs_effective, res.df_model - 1)
+    assert_allclose(actual_aic, res_glm.aic)
+    # TODO: Why does AIC match but BIC does not match?
```

Comment:
#1733 BIC uses a different definition specific to GLM.
And because of my inertia/reluctance with breaking backwards compatibility it still hasn't been changed.

Reply:
Makes sense!